backend/core/src/main/java/org/sonarsource/sonarlint/core/SonarCloudActiveEnvironment.java

```diff
@@ -20,28 +20,75 @@
 package org.sonarsource.sonarlint.core;
 
 import java.net.URI;
+import java.util.Map;
+
+import static org.apache.commons.lang.StringUtils.removeEnd;
 
 public class SonarCloudActiveEnvironment {
 
-  public static final URI PRODUCTION_URI = URI.create("https://sonarcloud.io");
+  public static final URI PRODUCTION_EU_URI = URI.create("https://sonarcloud.io");
+  public static final URI PRODUCTION_US_URI = URI.create("https://us.sonarcloud.io");
+  public static final URI WS_EU_ENDPOINT_URI = URI.create("wss://events-api.sonarcloud.io/");
+  public static final URI WS_US_ENDPOINT_URI = URI.create("wss://events-api.us.sonarcloud.io/");
 
   public static SonarCloudActiveEnvironment prod() {
-    return new SonarCloudActiveEnvironment(PRODUCTION_URI, URI.create("wss://events-api.sonarcloud.io/"));
+    return new SonarCloudActiveEnvironment();
   }
 
-  private final URI uri;
-  private final URI webSocketsEndpointUri;
+  private final Map<SonarCloudRegion, SonarQubeCloudUris> uris;
+
+  public SonarCloudActiveEnvironment() {
+    var euUris = new SonarQubeCloudUris(PRODUCTION_EU_URI, WS_EU_ENDPOINT_URI);
+    var usUris = new SonarQubeCloudUris(PRODUCTION_US_URI, WS_US_ENDPOINT_URI);
+    this.uris = Map.of(SonarCloudRegion.EU, euUris, SonarCloudRegion.US, usUris);
+  }
 
   public SonarCloudActiveEnvironment(URI uri, URI webSocketsEndpointUri) {
-    this.uri = uri;
-    this.webSocketsEndpointUri = webSocketsEndpointUri;
+    this.uris = Map.of(SonarCloudRegion.EU, new SonarQubeCloudUris(uri, webSocketsEndpointUri));
```

Comment:
This constructor is used if clients provide a custom URI. The risk here is that we don't define the US region in the map. Should we allow clients to also customize US URIs?

Reply:
Yes, this will be implemented later on as part of https://sonarsource.atlassian.net/browse/SLCORE-1151